### PR TITLE
Vps 169/group scenario last scene

### PR DESCRIPTION
**Pull request by harbassan**
## Describe the issue

When a user wants to play multiplayer, there is no architecture in place to allow them to be in sync with their group members.

## Describe the solution

- Refactored the previous code to make it easier to add more onto it
- Created a multiplayer component that adds logic for updating a group's current scene in the backend upon scene change.
- Created a `PlayScenarioResolver` component with logic that fetches group data for a user from the backend, and depending on the result navigates to either a single player, or multiplayer route with the correct current scene.
- Created the logic in the backend for the above functionality.

## Risk

Because this is a refactor, minor changes will probably need to made for other prs relating to the frontend of the single/multi play scenario page.

## Definition of Done

- [ ] Code peer-reviewed
- [ ] Wiki Documentation is written and up to date
- [ ] Unit tests written and passing
- [ ] Integration tests written and passing
- [ ] Continuous Integration build passing
- [ ] Acceptance criteria met
- [ ] Deployed to production environment

## Reviewed By

Who reviewed your PR - for commit history once merged

### CODE REVIEW

**Comment by harbassan** [comment]
Can I delete these tests, they're gonna need to be fully rewritten

File: frontend/src/containers/pages/PlayScenarioPage/PlayScenarioResolver.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import axios from "axios";
+
+import AuthenticationContext from "context/AuthenticationContext";
+import useGraph from "hooks/useGraph";
+
+import PlayScenarioPage from "./PlayScenarioPage";
+import PlayScenarioPageMulti from "./PlayScenarioPageMulti";
+
+// TODO: move this somewhere else and add error handling
+async function get(url, userIdToken) {
```
Comment (wjin-lee) [inline frontend/src/containers/pages/PlayScenarioPage/PlayScenarioResolver.jsx:18]: @JordanBlenn We should make a ticket to make sure we don't forget about this :)

**Review by wjin-lee — CHANGES_REQUESTED** [review]
Extremely clean refactor Hartej! I appreciate your effort in fully understanding this spaghetti of code and making everything much cleaner.

Couple jest snapshots need to be updated due to some changed imports it seems like, but other than that - extremely well done!

(give me a ping when you've regenerated the snapshots, and I'll approve it asap)

File: frontend/src/containers/pages/PlayScenarioPage/PlayScenarioResolver.jsx (hunk 1 above)
Comment (JordanBlenn) [reply frontend/src/containers/pages/PlayScenarioPage/PlayScenarioResolver.jsx:18]: Added a placeholder ticket to the backlog but I'll need you to fill out the details if that's all good :)

**Review by JordanBlenn — CHANGES_REQUESTED** [review]
Same as Jin - great job! Just resolve the paths for the test functions and should be all good

**Review by JordanBlenn — APPROVED** [review]
Great job!

**Review by wjin-lee — APPROVED** [review]
lgtm!

**Pull request merged**